SwedbankPaySDK/Classes/SwedbankPaySDK+Extensions/MerchantBackendConfiguration.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import Foundation
+import WebKit
+import Alamofire
+
+private let callbackURLTypeKey = "com.swedbank.SwedbankPaySDK.callback"
+
+public extension SwedbankPaySDK {
+    /// A SwedbankPaySDKConfiguration for integrating with a backend
+    /// implementing the Merchant Backend API.
+    struct MerchantBackendConfiguration: SwedbankPaySDKConfiguration {
+        internal let api: MerchantBackendApi
+        internal let rootLink: RootLink
+        
+        /// The url of the Merchant Backend
+        public var backendUrl: URL {
+            rootLink.href
+        }
+        public let callbackScheme: String
+        
+        let additionalAllowedWebViewRedirects: [WebViewRedirect]?
+        
+        /// Initializer for `SwedbankPaySDK.Configuration`
+        /// - parameter backendUrl: backend URL
+        /// - paramerer callbackScheme: A custom scheme for callback urls. This scheme must be registered to your app.
+        ///                             If nil, the Info.plist will be searched for a URL type
+        ///                             with a com.swedbank.SwedbankPaySDK.callback property having
+        ///                             a Boolean type and a YES value.
+        /// - parameter headers: HTTP Request headers Dictionary in a form of 'apikey, access token' -pair
+        /// - parameter domainWhitelist: Optional array of domains allowed to be connected to; defaults to `backendURL` if nil
+        /// - parameter certificatePins: Optional array of domains for certification pinning, matched against any certificate found anywhere in the app bundle
+        public init(backendUrl: URL, callbackScheme: String? = nil, headers: Dictionary<String, String>?, domainWhitelist: [WhitelistedDomain]? = nil, pinPublicKeys: [PinPublicKeys]? = nil, additionalAllowedWebViewRedirects: [WebViewRedirect]? = nil) {
+            let session = MerchantBackendConfiguration.makeSession(pinPublicKeys: pinPublicKeys)
+            self.init(session: session, backendUrl: backendUrl, callbackScheme: callbackScheme, headers: headers, domainWhitelist: domainWhitelist, additionalAllowedWebViewRedirects: additionalAllowedWebViewRedirects)
+        }
+        
+        internal init(session: Alamofire.Session, backendUrl: URL, callbackScheme: String?, headers: Dictionary<String, String>?, domainWhitelist: [WhitelistedDomain]?, additionalAllowedWebViewRedirects: [WebViewRedirect]?) {
```

Comment:
Codacy found an issue: [Shorthand syntactic sugar should be used, i.e. [String: Int] instead of Dictionary<String, Int>.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-ios/pullRequest?prid=6260585)